x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -836,7 +838,17 @@ private Join resolveLookupJoin(LookupJoin join, AnalyzerContext context) {
                     rightKeys = resolveUsingColumns(join.config().rightFields(), join.right().output(), "right");
                 }
                 config = new JoinConfig(type, leftKeys, rightKeys, joinOnConditions);
-                return new LookupJoin(join.source(), join.left(), join.right(), config, join.isRemote());
+                boolean remote = context.indexResolution()
```

Comment:
I wonder if it wouldn't make sense to have this bit captured somewhere e.g. in `IndexResolution` class? `IndexResolution` index set is immutable, as I understand, and we know upfront whether or not there are remote indices there as soon as it is created. Maybe then this should be the responsibility of this class to tell it, instead of doing this pattern in every place that needs to know (and there probably will be more places)?

Reply:
I refactored this a bit, now the calculation is delegated to the single components and the "remote" info in Join is nullable.